salt/utils/versions.py

```diff
@@ -110,14 +108,29 @@ def warn_until(
                                 issued. When we're only after the salt version
                                 checks to raise a ``RuntimeError``.
     """
-    if not isinstance(version, (tuple, str, salt.version.SaltStackVersion)):
+    if not isinstance(
+        version,
+        (tuple, int, str, salt.version.SaltVersion, salt.version.SaltStackVersion),
+    ):
```

Comment:
**question** would it make more sense to turn this into an `else` case? Otherwise we re-do `isinstance` checks.

Reply:
We still have to do all of the other `isinstance` checks, adding an `else` would not remove them...
So, why do it?

Reply:
That's what I'm saying... we're doing it here, and then we're doing it again in each of the cases.

For example - if it's a SaltStackVersion then here it would check tuple, int, str, SaltVersion, all failing, and then pass the SSV.

Then it would do *another* int check, another SV check, another tuple check, and another str check.

It's probably not a huge deal - but if we can simply do the checks one time, shouldn't we?

Reply:
Ah, so, what you're saying is do each of the checks first, then an else would replace the `if not`?

Reply:
Updated